### PR TITLE
Use gh-pages as P2 repository, take II

### PR DESCRIPTION
**Pull request by glhez**
Hello,
I've reworked my initial PR https://github.com/ddekany/jbosstools-freemarker/pull/8 

The checkout task does shallow clone and fetch only the last commit by default.
Now that I instructed it to fetch all, this now works using Java 8 without any higher version of Eclipse.

As a result, I've put the repo to https://glhez.github.io/jbosstools-freemarker/java8 instead of the root.
Another alternative would be to use another repository avoiding the need to download the jar each time.

The target platform is however using latest because I don't know the actual target platform.

### CODE REVIEW

File: releng/freemarker-ide.tpd [hunk 1]
```diff
@@ -0,0 +1,36 @@
+target "Freemarker IDE Target Platform"
+
+with requirements, source, configurePhase
+
+location "https://download.eclipse.org/releases/2021-06" {
```
Comment (ddekany) [inline releng/freemarker-ide.tpd]: Let's try to target https://download.eclipse.org/releases/2019-06/. Or even older, but 2 years old is generous enough, I think.

And then the fitting JBoss Tools parent version will be this: 4.12.0.Final-SNAPSHOT
I will try to test this on actual 2019-06 Eclipse... as I can find time, but maybe today.

**Comment by glhez** [comment]
The tpd is for building the target platform in case you do that in Eclipse.
I've used the P2 repositories mentioned by Tycho.

File: releng/freemarker-ide.target [hunk 2]
```diff
@@ -23,14 +23,11 @@
       <unit id="org.junit" version="0.0.0"/>
       <unit id="org.eclipse.ui.workbench" version="0.0.0"/>
       <unit id="org.eclipse.ui.ide.application" version="0.0.0"/>
-      <repository location="https://download.eclipse.org/releases/2021-06"/>
-    </location>
-    <location includeMode="planner" includeAllPlatforms="false" includeSource="true" includeConfigurePhase="true" type="InstallableUnit">
-      <repository location="https://download.eclipse.org/eclipse/updates/4.20"/>
+      <repository location="http://download.jboss.org/jbosstools/targetplatforms/jbosstoolstarget/4.71.1.Final/REPO"/>
```
Comment (ddekany) [inline releng/freemarker-ide.target:26]: 2019-06 is https://download.jboss.org/jbosstools/targetplatforms/jbosstoolstarget/4.12.0.Final-SNAPSHOT/REPO/, I think.

File: releng/freemarker-ide.target [hunk 3]
```diff
@@ -23,14 +23,11 @@
...
     </location>
     <location includeMode="planner" includeAllPlatforms="false" includeSource="true" includeConfigurePhase="true" type="InstallableUnit">
       <unit id="org.jboss.tools.tests" version="0.0.0"/>
-      <repository location="https://download.jboss.org/jbosstools/photon/snapshots/builds/jbosstools-base_master/latest/all/repo"/>
+      <repository location="http://download.jboss.org/jbosstools/oxygen/snapshots/builds/jbosstools-base_master/latest/all/repo"/>
```
Comment (ddekany) [inline releng/freemarker-ide.target:30]: I don't know what all these are (the most important is the Maven build for now anyway), but why oxygen specifically?

**Comment by ddekany** [comment]
Anyway... Thanks for the work on this! Seems to work on Eclipse 2019-06, so I merge, and will adjust dependency versions.

**Pull request merged**